### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Flodder23**
Added presets and exit and clear board buttons.

Would be nice to work out why presets won't place when game is not paused before I merge.

### CODE REVIEW

File: Game of Game of Life.py [hunk 1]
```diff
@@ -70,14 +69,33 @@ def check_user_input(paused):
...
                 paused = True
+        if event.type == pygame.QUIT or pygame.key.get_pressed()[pygame.K_ESCAPE]:
+            pygame.quit()
+            import sys
+            sys.exit(0)
+        if pygame.key.get_pressed()[pygame.K_1]:
+            board = preset.place(board, 1, a, b)
+        if pygame.key.get_pressed()[pygame.K_2]:
+            board = preset.place(board, 2, a, b)
+        if pygame.key.get_pressed()[pygame.K_3]:
```
Comment (max-kov) [inline Game of Game of Life.py]: make this into one line of code with https://www.pygame.org/docs/ref/key.html#pygame.key.name

File: preset.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        if a + 2 >= config.Width + config.Cushion:
+            a = config.Width + config.Cushion - 3
+        if b + 2 >= config.Height + config.Cushion:
+            b = config.Height + config.Cushion - 3
+        board[a][b].kill()
+        board[a + 1][b].birth(config.Square)
+        board[a + 2][b].kill()
+        board[a][b + 1].kill()
+        board[a + 1][b + 1].kill()
+        board[a + 2][b + 1].birth(config.Square)
```
Comment (max-kov) [inline preset.py]: looks awful, i'd look into making the board a numpy array and using this https://stackoverflow.com/questions/26975769/modify-a-particular-row-column-of-a-numpy-array

**Comment by max-kov** [comment]
`Cushion=5   #C 10 How far the board extends beyond the visible amount` bad solution for the problem of the game border, i'd make the game register the outermost cells as always alive, that way when a cell appears, it will die of overpopulation, or just don't spawn any cells there. The cushion variable makes it so that every time you need the number of cells in the game, you have to recalculate it again.

File: preset.py (hunk 2 above)
Comment (max-kov) [reply preset.py]: Not saying that it is possible, but i;d look into possible solutions

**Comment by max-kov** [comment]
```import math as m``` in config file, but ```import math as maths``` in the main file, not good. I'd switch to ```import math```

**Comment by max-kov** [comment]
You don't need the `Dead=0` variable in the config file, just make a boolean variable every time you need to make a cell `Cell.IsAlive = True`

**Comment by max-kov** [comment]
```Edge=m.ceil(Size/7)    #C Size/15 The gap between each cell```
Magic values. 7? Why not 8?

File: Game of Game of Life.py (hunk 1 above)
Comment (joegibby) [reply Game of Game of Life.py]: nice spot, went from 14 lines to 3.

**Comment by joegibby** [comment]
@harleygn

File: preset.py (hunk 2 above)
Comment (joegibby) [reply preset.py]: Being able to modify a row/column won't help, most of the presets annoyingly don't follow any pattern at all so as far as I am aware this is the best way to place them.

**Pull request merged**